src/server/lib/ab.ts

```diff
@@ -60,9 +72,83 @@ export const selectVariant = <V extends Variant, T extends Test<V>>(test: T, mvt
     return selectWithSeed(mvtId, seed, test.variants);
 };
 
-export const selectVariantNonSticky = <V extends Variant, T extends Test<V>>(test: T): V => {
-    const index = Math.floor(Math.random() * test.variants.length);
-    return test.variants[index];
+const selectVariantWithMethodology = <V extends Variant, T extends Test<V>>(
+    test: T,
+    mvtId: number,
+    banditData: BanditData[],
+    methodology: Methodology,
+): V | undefined => {
+    if (methodology.name === 'EpsilonGreedyBandit') {
+        return selectVariantUsingEpsilonGreedy(banditData, test, methodology.epsilon);
+    }
+    return selectVariantUsingMVT<V, T>(test, mvtId);
+};
+
+const addMethodologyToTestName = (testName: string, methodology: Methodology): string => {
+    if (methodology.name === 'EpsilonGreedyBandit') {
+        return `${testName}_EpsilonGreedyBandit-${methodology.epsilon}`;
+    } else {
+        return `${testName}_ABTest`;
+    }
+};
+
+/**
+ * Selects a variant from the test based on any configured methodologies.
+ * Defaults to an AB test.
+ */
+export const selectVariant = <V extends Variant, T extends Test<V>>(
+    test: T,
+    mvtId: number,
+    banditData: BanditData[],
+): { test: T; variant: V } | undefined => {
+    if (test.methodologies && test.methodologies.length === 1) {
+        // Only one configured methodology
+        const variant = selectVariantWithMethodology<V, T>(
+            test,
+            mvtId,
+            banditData,
+            test.methodologies[0],
+        );
+        if (variant) {
+            return {
+                test,
+                variant,
+            };
+        }
+    } else if (test.methodologies) {
+        // More than one methodology, pick one of them using the mvt value
```

Comment:
How does this know to spilt it 50/50?

Reply:
The `getRandomNumber` function returns a value based on the user's `mvtId`.
The `% test.methodologies.length` part divides all values between the methodologies. So if there are 2 methodologies then they'll be split 50/50.
This is similar to the logic in `selectWithSeed` elsewhere in this file.